datalad/core/local/status.py

```diff
@@ -495,3 +495,24 @@ def custom_result_summary_renderer(results):  # pragma: more cover
                for r in results):
             from datalad.ui import ui
             ui.message("nothing to save, working tree clean")
+
+        # ### Report on possibly cleanable temporary locations
+
+        # Infer the dataset we ran on from the results and get original
+        # recursive (+limit) options:
+        refds = set(r['refds'] for r in results)
+        assert len(refds) == 1
```

Comment:
If for whatever reason the assertion is wrong, it will crash the process. I think this is a bit extreme for a convenience report. Maybe just `return`?

Reply:
True. Actually only wanted to see whether that holds throughout the test battery. Replacement kinda depends on the outcome. Either `return` or failures point to how I got it wrong and can adjust.

Reply:
Changed to just issue a debug log message and proceed with the first discovered refds (which now is only needed to retrieve a config).